ts/client/scripts/risk.ts

```diff
@@ -0,0 +1,574 @@
+  const response = await fetch(url);
+
+  try {
+    let res = await response.json();
+    res = res.data[0];
+    return {
+      outAmount: parseFloat(res.outAmount),
+      priceImpactPct: parseFloat(res.priceImpactPct),
+    };
+  } catch (e) {
+    console.log(url);
+    console.log(e);
+    throw e;
+  }
+}
+
+async function computePriceImpactForLiqor(
+  group: Group,
+  mangoAccounts: MangoAccount[],
+  healthThresh: number,
+  title: string,
+  csvSuffix: string,
+): Promise<void> {
+  // Filter mango accounts below a certain health ration threshold
+  const mangoAccountsWithHealth = mangoAccounts
+    .map((a: MangoAccount) => {
+      return {
+        account: a,
+        health: a.getHealth(group, HealthType.liquidationEnd),
+        healthRatio: a.getHealthRatioUi(group, HealthType.liquidationEnd),
+        liabs: toUiDecimalsForQuote(
+          a.getLiabsValue(group, HealthType.liquidationEnd),
+        ),
+      };
+    })
+    .filter((a) => a.healthRatio < healthThresh);
+
+  const table = new Table({
+    columns: [
+      { name: 'Coin', alignment: 'right' },
+      { name: 'Oracle Price', alignment: 'right' },
+      { name: 'On-Chain Price', alignment: 'right' },
+      { name: 'Future Price', alignment: 'right' },
+      // { name: 'V4 Soft Limit', alignment: 'right' },
+      { name: 'V4 Liq Fee', alignment: 'right' },
+      { name: 'Liabs', alignment: 'right' },
+      { name: 'Liabs slippage', alignment: 'right' },
+      { name: 'Assets Sum', alignment: 'right' },
+      { name: 'Assets Slippage', alignment: 'right' },
+      // { name: 'Jup Day Volume', alignment: 'right' },
+    ],
+  });
+
+  const fileName = `/tmp/${
+    new Date().toISOString().split('T')[0]
+  }-${csvSuffix}-price_impact.csv`;
+  const csvFile = fs.createWriteStream(fileName);
+  const stream = format({ headers: true });
+  stream.pipe(csvFile);
+  stream.write([
+    'Coin',
+    'Oracle Price',
+    'On-Chain Price',
+    'Future Price',
+    // 'V4 Soft Limit',
+    'V4 Liq Fee',
+    'Liabs',
+    'Liabs slippage',
+    'Assets Sum',
+    'Assets Slippage',
+    // 'Jup Day Volume',
+  ]);
+
+  // High level solana defi stats
+  const response = await fetch('https://cache.jup.ag/stats/day');
+  const res = await response.json();
+
+  const USDC_MINT = 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v';
+  const usdcBank = group.getFirstBankByMint(new PublicKey(USDC_MINT));
+
+  // For each token
+  for (const banks of Array.from(group.banksMapByMint.values())) {
+    // TODO USDC
+    const bank = banks[0];
+
+    const onChainPrice = (
+      await (
+        await fetch(`https://price.jup.ag/v4/price?ids=${bank.mint}`)
+      ).json()
+    )['data'][bank.mint.toBase58()]['price'];
+
+    if (bank.reduceOnly > 0 || bank.tokenIndex === usdcBank.tokenIndex) {
+      continue;
```

Comment:
minor: could do it above onChainPrice fetching

Is it really right to skip reduceOnly tokens? Like DUAL for instance could still be liquidated...